frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java

```diff
@@ -76,6 +80,47 @@ public WorkerState getState() {
         return state;
     }
 
+    public String getGpuUsage() {
+        Process process;
+        StringBuffer gpuUsage = new StringBuffer();
+        if (gpuId >= 0) {
+            try {
+                process =
+                        Runtime.getRuntime()
+                                .exec(
+                                        "nvidia-smi -i "
```

Comment:
NIT: This seems to be nvidia specific. Can you investigate if we can do this in a more generic way using some opengl java library?

Reply:
@maaquib: Agreed, we should use a more generic approach. However, we are currently supporting NVIDIA-CUDA only. I will add a TODO for this.

Reply:
Done.

Reply:
@harshbafna Instead of just a TODO can you also add an issue to the backlog

Reply:
Done. Created #740.